src/member-api.js

```diff
@@ -51,19 +56,38 @@ export class MemberApi extends TypedEmitter {
    *
    * @param {Object} opts
    * @param {import('./capabilities.js').RoleId} opts.roleId
+   * @param {string} [opts.roleName]
+   * @param {string} [opts.roleDescription]
    * @param {number} [opts.timeout]
    *
    * @returns {Promise<import('./generated/rpc.js').InviteResponse_Decision>}
    */
-  async invite(deviceId, { roleId, timeout }) {
-    const projectId = projectKeyToId(this.#projectKey)
+  async invite(deviceId, { roleId, roleName, roleDescription, timeout }) {
+    if (!(roleId === MEMBER_ROLE_ID || roleId === COORDINATOR_ROLE_ID)) {
+      throw new Error('Invalid role id')
+    }
+
+    const { name: deviceName } = await this.getById(this.#ownDeviceId)
+
+    // since we are always getting #ownDeviceId,
+    // this should never throw (see comment on getById), but it pleases ts
+    if (!deviceName) throw new Error('Invalid deviceName')
 
+    if (!roleName) throw new Error('Invalid roleId')
```

Comment:
This shouldn't be needed (and the error message is misleading). It's also causing some tests to fail when they shouldn't (since the roleName is truly optional and has a fallback)

```suggestion
```

Reply:
heh, yeah, I pushed in a rush :face_in_clouds: